docs/source/release/0.59.0-notes.rst

```diff
@@ -0,0 +1,527 @@
+* PR `#9296 <https://github.com/numba/numba/pull/9296>`_: Fix bug when axis is negative and check when axis is invalid (`guilhermeleobas <https://github.com/guilhermeleobas>`_)
+* PR `#9301 <https://github.com/numba/numba/pull/9301>`_: Release Notes 0.58.1 for main (`esc <https://github.com/esc>`_)
+* PR `#9302 <https://github.com/numba/numba/pull/9302>`_: add missing backtick to example git tag command (`esc <https://github.com/esc>`_)
+* PR `#9303 <https://github.com/numba/numba/pull/9303>`_: Add category to warning (`kkokkoros <https://github.com/kkokkoros>`_)
+* PR `#9307 <https://github.com/numba/numba/pull/9307>`_: Upgrade to cloudpickle 3.0.0 (`sklam <https://github.com/sklam>`_)
+* PR `#9308 <https://github.com/numba/numba/pull/9308>`_: Fix typo in azure ci script (`sklam <https://github.com/sklam>`_)
+* PR `#9309 <https://github.com/numba/numba/pull/9309>`_: Continue #9044, prevent compilation on the CPU target when listing tests. (`stuartarchibald <https://github.com/stuartarchibald>`_ `apmasell <https://github.com/apmasell>`_)
+* PR `#9310 <https://github.com/numba/numba/pull/9310>`_: Remove Python 3.8 support. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#9315 <https://github.com/numba/numba/pull/9315>`_: Fix support for LOAD_FAST_AND_CLEAR (`sklam <https://github.com/sklam>`_)
+* PR `#9318 <https://github.com/numba/numba/pull/9318>`_: GPU CI: Test with Python 3.9-3.12 (`gmarkall <https://github.com/gmarkall>`_)
+* PR `#9325 <https://github.com/numba/numba/pull/9325>`_: Fix GPUCI (`gmarkall <https://github.com/gmarkall>`_)
+* PR `#9326 <https://github.com/numba/numba/pull/9326>`_: Add docs for LOAD_FAST_AND_CLEAR changes (`sklam <https://github.com/sklam>`_)
+* PR `#9330 <https://github.com/numba/numba/pull/9330>`_: Refactor and remove legacy APIs/testing internals. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#9331 <https://github.com/numba/numba/pull/9331>`_: Fix Syntax and Deprecation Warnings from 3.12. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#9334 <https://github.com/numba/numba/pull/9334>`_: Fix parfor reduction issue with Python 3.12. (`DrTodd13 <https://github.com/DrTodd13>`_)
+* PR `#9335 <https://github.com/numba/numba/pull/9335>`_: Add validation capability for user generated towncrier .rst files. (`kc611 <https://github.com/kc611>`_)
+* PR `#9336 <https://github.com/numba/numba/pull/9336>`_: Add TargetLibraryInfo pass to CPU LLVM pipeline. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#9337 <https://github.com/numba/numba/pull/9337>`_: Revert #8583 which skip tests due to M1 RuntimeDyLd Assertion error (`sklam <https://github.com/sklam>`_)
+* PR `#9341 <https://github.com/numba/numba/pull/9341>`_: Add configuration variable to force llvmlite memory manager on / off (`gmarkall <https://github.com/gmarkall>`_)
+* PR `#9342 <https://github.com/numba/numba/pull/9342>`_: Fix flake8 checks for v6.1.0 (`gmarkall <https://github.com/gmarkall>`_)
```

Comment:
Was this generated against current HEAD of `main` (48fe645)? PRs merged recently, e.g. #9360 are missing?